src/lib.rs

```diff
@@ -167,10 +167,14 @@ pub enum TweeningType {
     Once,
     /// Loop the animation indefinitely, restarting from the start each time the end is reached.
     Loop,
+    /// [Loop][TweeningType::Loop]s N times.
+    LoopTimes(u32),
```

Comment:
I've come up with the same conclusion that we need something like that while working on making `Sequence` loopable, but I think there's a better design than that. Adding even more tweening types just makes handling them all more complex.

Instead, we can decouple the concepts of _repeat count_ on one side, and _repeat mode_ on the other side.
- The repeat count is self-explanatory.
- The repeat mode is simply `Normal` or `Mirrored`, describes how the playback direction changes (or not) when reaching the end of the current playback cycle/loop.

All existing `TweeningType` enum values, as well as the ones added in this PR, can be described in terms of those two quantities. And I'm pretty sure this makes some operations easier to write internally, as we sometimes care about the repeat count or the mirroring, but rarely both at once.

Note that although I'm grateful for the contribution, and always open to them, it's probably better to start a discussion on Discord or open a GitHub issue to discuss this kind of change before jumping into an implementation. Same for the speed feature one.

Reply:
Oh interesting, so does that mean PingPong can be deleted since you could combine Loop with Mirrored?

I'll let the speed PR settle before working on the others. As for why I went straight to PRs, I needed all this code for my game and am already using it. :grin:

Reply:
Yes indeed, `Once` is "Repeat forward 1 time" (`repeat=Times(1), mode=Normal`), `Loop` is "Repeat forward infinite times" (`repeat=Infinite, mode=Normal`), and `PingPong` is "Repeat mirrored infinite times" (`repeat=Infinite, mode=Mirrored`).

Reply:
That's super smart, I love it :heart_eyes: